hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/protocolPB/OMPBHelper.java

```diff
@@ -72,24 +81,19 @@ public static OzoneAclInfo convertOzoneAcl(OzoneAcl acl) {
     default:
       throw new IllegalArgumentException("ACL type is not recognized");
     }
-    OzoneAclInfo.OzoneAclRights aclRights;
-    switch(acl.getRights()) {
-    case READ:
-      aclRights = OzoneAclRights.READ;
-      break;
-    case WRITE:
-      aclRights = OzoneAclRights.WRITE;
-      break;
-    case READ_WRITE:
-      aclRights = OzoneAclRights.READ_WRITE;
-      break;
-    default:
-      throw new IllegalArgumentException("ACL right is not recognized");
+    List<OzoneAclRights> aclRights = new ArrayList<>();
+
+    for(ACLType right: acl.getRights()) {
+      try {
+        aclRights.add(OzoneAclRights.valueOf(right.name()));
+      } catch (IllegalArgumentException iae) {
+        LOG.error("ACL:{} right is not recognized.", acl);
```

Comment:
This should be more than a LOG.Error? if we don't recognize an ACL shouldn't we return an error to the caller instead of ignoring and continue?

Reply:
Since this involves a list of acls IMO a single bad acl type should not halt the usage of remaining ones. But i am open to change if you think we should throw error at this point.

Reply:
Agree with @anuengineer based on offline discussion with @ajayydv . We should stop here instead of catch and let the parsing continue.